tensorflow_addons/layers/noisy_dense.py

```diff
@@ -208,23 +219,27 @@ def bias(self):
         if self.use_bias:
             return self.mu_bias + (self.sigma_bias * self.eps_bias)
 
+    def _scaled_noise(self, size, dtype):
+        x = tf.random.normal(shape=size, dtype=dtype)
+        return tf.sign(x) * tf.sqrt(tf.abs(x))
+
     def _reset_noise(self):
         """Create the factorised Gaussian noise."""
 
         dtype = self._compute_dtype_object
 
-        # Generate random noise
-        eps_i = tf.random.normal([self.last_dim, self.units], dtype=dtype)
-        eps_j = tf.random.normal(
-            [
-                self.units,
-            ],
-            dtype=dtype,
-        )
+        if self.use_factorised:
+            # Generate random noise
+            in_eps = self._scaled_noise([self.last_dim, 1], dtype=dtype)
+            out_eps = self._scaled_noise([1, self.units], dtype=dtype)
 
-        # Scale the random noise
-        self.eps_kernel = _scale_noise(eps_i) * _scale_noise(eps_j)
-        self.eps_bias = _scale_noise(eps_j)
+            # Scale the random noise
+            self.eps_kernel = tf.matmul(in_eps, out_eps)
+            self.eps_bias = out_eps[0]
+        else:
+            # generate independent variables
+            self.eps_kernel = tf.random.normal(shape=[self.last_dim, self.units], dtype=dtype)
+            self.eps_bias = tf.random.normal(shape=[self.units,], dtype=dtype)
```

Comment:
I probably should have done it like this originally, but now you could just put the noise scaling in an if statement (unless this is faster) and generate above since the method being used is generating the noise on 2 occasions.

Reply:
The noise if is self.use_factorised = True is generated with different shape and much faster then independent noise that is generated for (self.last_dim, self.units) instead of (self.last_dim, 1). The if statement is okey, because you must decide, which method use to generate smaller but combined noise with scale or use independent noise without scaling .... please look to original paper: https://arxiv.org/pdf/1706.10295.pdf